File: Packages/Flip54Core/Sources/Flip54Core/Exercise.swift
public enum Exercise: String, Codable, Hashable, Sendable {
    // Lower body
    case bodyweightSquat
    case lunge
    case jumpingSquat
    case gobletSquat

    // Upper body
    case pushUp
    case hinduPushUp
    case pullUp
    case bicepCurl
    case shoulderPress
    case tricepExtension

    // Total body
    case burpee
    case mountainClimber
    case thruster

    // Core
    case sitUp
    case russianTwist
    case weightedSitUp

    // Conditioning
    case jumpingJacks

    // Holds (Aces)
    case pushUpHold
    case deadHang
    case hollowBodyHold
    case wallSit
    case plank

    public var isHold: Bool {
        switch self {
        case .pushUpHold, .deadHang, .hollowBodyHold, .wallSit, .plank: return true
        default: return false
        }
    }

    public var displayName: String {
        switch self {
        case .bodyweightSquat: return "Body-weight Squats"
        case .lunge:           return "Lunges"
        case .jumpingSquat:    return "Jumping Squats"
        case .gobletSquat:     return "Goblet Squats"
        case .pushUp:          return "Push-ups"
        case .hinduPushUp:     return "Hindu Push-ups"
        case .pullUp:          return "Pull-ups"
        case .bicepCurl:       return "Bicep Curls"
        case .shoulderPress:   return "Shoulder Press"
        case .tricepExtension: return "Tricep Extensions"
        case .burpee:          return "Burpees"
        case .mountainClimber: return "Mountain Climbers"
        case .thruster:        return "Thrusters"
        case .sitUp:           return "Sit-ups"
        case .russianTwist:    return "Russian Twists"
        case .weightedSitUp:   return "Weighted Sit-ups"
        case .jumpingJacks:    return "Jumping Jacks"
        case .pushUpHold:      return "Push-up Hold"
        case .deadHang:        return "Dead Hang"
        case .hollowBodyHold:  return "Hollow Body Hold"
        case .wallSit:         return "Wall Sit"
        case .plank:           return "Plank"
        }
    }
}

File: android/core/src/main/kotlin/com/flip54/core/Exercise.kt
package com.flip54.core

import kotlinx.serialization.Serializable

@Serializable
enum class Exercise {
    // Lower body
    BODYWEIGHT_SQUAT, LUNGE, JUMPING_SQUAT, GOBLET_SQUAT,
    // Upper body
    PUSH_UP, HINDU_PUSH_UP, PULL_UP, BICEP_CURL, SHOULDER_PRESS, TRICEP_EXTENSION,
    // Total body
    BURPEE, MOUNTAIN_CLIMBER, THRUSTER,
    // Core
    SIT_UP, RUSSIAN_TWIST, WEIGHTED_SIT_UP,
    // Conditioning
    JUMPING_JACKS,
    // Holds (Aces)
    PUSH_UP_HOLD, DEAD_HANG, HOLLOW_BODY_HOLD, WALL_SIT, PLANK;

    val isHold: Boolean get() = when (this) {
        PUSH_UP_HOLD, DEAD_HANG, HOLLOW_BODY_HOLD, WALL_SIT, PLANK -> true
        else -> false
    }

    val displayName: String get() = when (this) {
        BODYWEIGHT_SQUAT  -> "Body-weight Squats"
        LUNGE             -> "Lunges"
        JUMPING_SQUAT     -> "Jumping Squats"
        GOBLET_SQUAT      -> "Goblet Squats"
        PUSH_UP           -> "Push-ups"
        HINDU_PUSH_UP     -> "Hindu Push-ups"
        PULL_UP           -> "Pull-ups"
        BICEP_CURL        -> "Bicep Curls"
        SHOULDER_PRESS    -> "Shoulder Press"
        TRICEP_EXTENSION  -> "Tricep Extensions"
        BURPEE            -> "Burpees"
        MOUNTAIN_CLIMBER  -> "Mountain Climbers"
        THRUSTER          -> "Thrusters"
        SIT_UP            -> "Sit-ups"
        RUSSIAN_TWIST     -> "Russian Twists"
        WEIGHTED_SIT_UP   -> "Weighted Sit-ups"
        JUMPING_JACKS     -> "Jumping Jacks"
        PUSH_UP_HOLD      -> "Push-up Hold"
        DEAD_HANG         -> "Dead Hang"
        HOLLOW_BODY_HOLD  -> "Hollow Body Hold"
        WALL_SIT          -> "Wall Sit"
        PLANK             -> "Plank"
    }
}
